scripts/ibmcloud_cli_utils.sh

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+function ibmcloud_login ()
+{
+    region="${1}"
+    rg="${2}"
+    api_visibility="${3}"
+
+    # Target the resource group and region if passed in
+    args=(login --quiet)
+    if [[ -n "${rg}" && "${rg}" != "UNSET" ]]; then
+        args+=(-g "${rg}")
+    fi
+    if [[ -n "${region}" && "${region}" != "UNSET" ]]; then
+        args+=(-r "${region}")
+    else
+        args+=(--no-region)
+    fi
+    # change api to private if visibility not public
+    if [[ -n "${api_visibility}" && "${api_visibility}" != "UNSET" && "${api_visibility}" != "public" ]]; then
+        args+=(-a "private.cloud.ibm.com" --vpc)
+    fi
+
+    # ibmcloud login (with 3 retry attempts)
+    total_attempts=3
+    i=0
+    wait=3
+    until [ "$i" -ge $total_attempts ]; do
+    ibmcloud "${args[@]}" && break
+    i=$((i+1))
+    if [ "$i" = $total_attempts ]; then
+        echo "Maximum login attempts reached. Giving up!" >&2
+        exit 1
+    else
+        echo "Error, retrying in ${wait} secs .." >&2
+        sleep ${wait}
+    fi
+    done
+}
+
+function setup_temp_config_home ()
+{
+    if [[ -z "${IBMCLOUD_HOME:-}" ]]; then
+        old_home="${HOME}"
+    else
+        old_home="${IBMCLOUD_HOME}"
+    fi
+
+    # Create a temporary home for CLI config, used by both login and further commands.
+    # This ensures config separation between various terraform provision blocks on the same machine.
+    ibmcloud_config_home=$(mktemp -d)
```

Comment:
we should probably be cleaning up the temp directory after? But I guess it would have to be done after the scripts have completed, which is outside the scope of this function. Perhaps there is a cli command to get current home, and delete it (if its not the default home)?

Reply:
You are right, I forgot the cleanup. It should be easy since I have it in an ENV already I can just remove the whole directory at the end of the script. However, if the script errors, it won't remove. I could look into doing a bash-style error handling block to remove it there, if that is important.

One thing I will do though is ignore an error in the removal, I don't want that to trigger error in terraform.

Another note is that `mktemp` will always create the folder in the underlying OS official $TMPDIR location, so theoretically it will get cleaned up eventually.

Reply:
Latest version I've added a removal of the temp directory that will occur with normal script termination, and also if an error occurs anywhere (via bash `trap ERR` mechanic). I've tested error scenarios at every level and it seems to be getting caught and removed.